backend/app/datamodels/tables.js

```diff
@@ -1,72 +1,139 @@
-    table.olderTabs.push(table.currentTab);
-    table.currentTab = null;
-    table.tabOpen = false;
-    table.lastUpdated = Date.now();
+	if (!table.currentTab) {
+		throw new Error('No current tab for the table');
+	}
+	const currentTab = table.currentTab;
+	table.olderTabs.push(table.currentTab);
+	table.currentTab = null;
+	table.tabOpen = false;
+	table.lastUpdated = Date.now();
 
-    await table.save({ session });
+	await table.save({ session });
 
-    return { table, tabId: currentTab }
+	return { table, tabId: currentTab };
 };
 
 const getTableById = async (tableId) => {
-    const table = await Tables.findById(tableId).populate({
-        path: 'currentTab',
-        populate: {
-            path: 'orders',
-            populate: {
-                path: 'items',
-                populate: {
-                    path: 'recipe'
-                }
-            }
-        }
-    }).exec();
-
-    if (!table) {
-        throw new Error('Table not found');
-    }
-
-    return table;
+	const table = await Tables.findById(tableId).populate({
+		path: 'currentTab',
+		populate: {
+			path: 'orders',
+			populate: {
+				path: 'items',
+				populate: {
+					path: 'recipe'
+				}
+			}
+		}
+	}).exec();
+
+	if (!table) {
+		throw new Error('Table not found');
+	}
+
+	return table;
 };
 
 const getTableByRestaurantId = async (restaurantId) => {
-    const table = await Tables.find({ restaurantId }).populate({
-        path: 'currentTab',
-        populate: {
-            path: 'orders',
-            populate: {
-                path: 'items',
-                populate: {
-                    path: 'recipe'
-                }
-            }
-        }
-    }).exec();
-
-    if (!table) {
-        throw new Error('Table not found');
-    }
-
-    return table;
+	const table = await Tables.find({ restaurantId }).populate({
+		path: 'currentTab',
+		populate: {
+			path: 'orders',
+			populate: {
+				path: 'items',
+				populate: {
+					path: 'recipe'
+				}
+			}
+		}
+	}).exec();
+
+	if (!table) {
+		throw new Error('Table not found');
+	}
+
+	return table;
+};
+
+const getTableStatus = async (tableId) => {
+	const table = await Tables.findById(tableId);
+
+	if (!table) {
+		throw new Error('Table not found');
+	}
+
+	if (table.locked) {
+		throw new Error('Table is locked');
+	}
+
+	return true;
 };
 
+const populateTableData = async (table) => {
+	return await table.populate({
+		path: 'currentTab',
+		populate: {
+			path: 'orders',
+			options: { sort: { createdAt: -1 } },
+			populate: {
+				path: 'items',
+				populate: {
+					path: 'recipe'
+				}
+			}
+		}
+	}).execPopulate();
+};
 
 module.exports = {
-    closeTable,
-    getTableByRestaurantId,
-    getTableById,
+	getTable,
+	getRandomTable,
+	closeTable,
+	getTableStatus,
+	getTableByRestaurantId,
+	getTableById,
 };
+
+// TODO: refactor the getTableById and getTable to only one function and make necceary changes in the frontend
\ No newline at end of file
```

Comment:
The code review reveals the following points:

- In the `getRandomTable` function, it may not be sufficient to just randomly fetch a table for the given restaurant without checking if the table is available. A better approach would be to fetch all available tables and then randomly select one from the list.
- The `getTable` function could benefit from a refactoring where checks for table availability and pin verification are separated into their own functions to promote better readability and easier testing.
- In the `closeTable` function, there could be an additional check to verify that the current tab for the table being closed is not empty before adding it to the older tabs array.
- The `populateTableData` function helps to load the table's pertinent data alongside its relevant dependencies. Still, the use of `execPopulate()` is unnecessary since it defaults to returning a Promise when called.
 
Furthermore, the author has provided a TODO note to merge the `getTableById` function with `getTable`. 

Overall, the code seems to be well structured, easy to read and comprehensive in the context of the application, providing good coverage of the necessary CRUD operations on Tables models.